### DISCUSSION TITLE
Generate CRS from PROJ.4 a/rf equivalent to ellps=WGS84

### DISCUSSION THREAD

**Discussion by djhoese** [pyproj4/pyproj#1284] [Q&A]
I have a case where I'm compare two CRS objects that should be considered equivalent but are not. Both are created from PROJ.4 dictionaries. One has `a` and `rf` defined:

```python
{'a': 6378137.0, 'lon_0': 0.0, 'h': 35786400.0, 'rf': 298.257223563, 'proj': 'geos', 'units': 'm', 'sweep': 'y'}
```

And is coming from a data file from a satellite instrument.

The other CRS is coming from a YAML configuration file:

```python
{'proj': 'geos',
    'lon_0': 0,
    'h': 35786400,
    'x_0': 0,
    'y_0': 0,
    'ellps': 'WGS84',
    'no_defs': None,
}
```

I've also tried adding `datum: 'WGS84'` to the above but the resulting datums in the CRS object aren't considered equivalent:

```
# from a/rf
DATUM["unknown",
    ELLIPSOID["WGS 84",6378137,298.257223563,
        LENGTHUNIT["metre",1,
            ID["EPSG",9001]]]]
```

```
# from yaml ellps
DATUM["Unknown based on WGS84 ellipsoid",
    ELLIPSOID["WGS 84",6378137,298.257223563,
        LENGTHUNIT["metre",1],
        ID["EPSG",7030]]]
```

```
# from yaml datum/ellps
DATUM["World Geodetic System 1984",
    ELLIPSOID["WGS 84",6378137,298.257223563,
        LENGTHUNIT["metre",1]],
    ID["EPSG",6326]]
```

Since the a and rf are coming from a data file's attribute I can't really confidently change them to datum/ellps. Does anyone have ideas how I could specify these to be considered equal?

**Comment by snowman2** [comment]
The logic for equality is handled upstream in PROJ. I think that you may have better luck asking the question there.

**Reply by djhoese** [reply]
Thanks. Makes sense. Do you have a good minimal example I could post to show this issue? Or maybe they'd be OK with a pyproj example for now?

**Reply by snowman2** [reply]
Here is some pseudo code:
```c
PJ_CONTEXT* context = proj_context_create();
PJ* pj1 = proj_create(
    context,
     "+proj=geos +a=6378137.0 +lon_0=0.0 +h=35786400.0 +rf=298.257223563 +units=m +sweep=y +type=crs"
);
PJ* pj2 = proj_create(
    context,
     "+proj=geos +lon_0=0 +h=35786400 +x_0=0 +y_0=0 +ellps=WGS84 +no_defs +type=crs"
);
proj_is_equivalent_to_with_ctx(
    context, pj1, pj2, PJ_COMP_EQUIVALENT
);
```

**Reply by djhoese** [reply]
https://github.com/OSGeo/PROJ/issues/3725
